pkg/util/fixtest/remove_duplicates.go

```diff
@@ -0,0 +1,65 @@
+package fixtest
+
+import (
+	"context"
+	"log"
+	"sort"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type Logger interface {
+	Printf(fmt string, args ...any)
+}
+
+// EnsureNoDuplicates removes projects with same name but different ID.
+// Atlas sometimes creates duplicate projects, we need our tests to defend
+// against that to avoid flaky tests
+func EnsureNoDuplicates(client *mongodbatlas.Client, logger Logger, projectName string) error {
+	found, err := listProjectsByName(client, projectName, 250)
+	if err != nil || len(found) <= 1 {
+		return err
+	}
+	log.Printf("Found more than one project with name %q", projectName)
+	keep, rest := selectProject(found)
+	log.Printf("Will keep project ID %s as %s and remove the rest", keep.ID, projectName)
+	return removeProjects(client, rest)
+}
+
+func listProjectsByName(client *mongodbatlas.Client, projectName string, pageSize int) ([]*mongodbatlas.Project, error) {
+	projects, _, err := client.Projects.GetAllProjects(
+		context.Background(),
+		&mongodbatlas.ListOptions{
```

Comment:
I think this is redundant. Atlas has already a limit of projects per Org and this function doesn't enable page navigation. 
```suggestion
		&mongodbatlas.ListOptions{},
```

Reply:
done in next change